modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/IgniteRpcClient.java

```diff
@@ -121,6 +120,10 @@ else if (err instanceof TimeoutException) // Translate timeout exception.
 
                 blockedMsgs.add(msgData);
 
+                if (timeoutMs > 0) {
+                    fut.orTimeout(timeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
Could you please elaborate why this is better? We still call `orTimeout()`, just later.

Reply:
It is a code for tests in which we block RAFT commands.